data/text.py

```diff
@@ -56,3 +59,16 @@ def match_like(field, search_query):
     escaped_query = _escape_wildcard(search_query)
     clause = NodeList(("%" + escaped_query + "%", SQL("ESCAPE '!'")))
     return Field.__pow__(field, clause)
+
+
+def regex_search(query, field, pattern, matches=True):
+    return query.where(field.regexp(pattern)) if matches else query.where(~field.regexp(pattern))
+
+
+def regex_sqlite(query, field, pattern, matches=True):
+    rows = query.execute()
+    return (
+        [row for row in rows if re.search(pattern, getattr(row, field.name))]
+        if matches
+        else [row for row in rows if not re.search(pattern, getattr(row, field.name))]
+    )
```

Comment:
Instead of iterating over all rows, will `like 'pattern'` work here?

Reply:
`like` uses a SQL specific way of pattern matching while we want to match via regex patterns. Currently sqlite doesn't support regex pattern matching

Reply:
yes, that makes sense